libeos-payg/manager.c

```diff
@@ -1152,6 +1320,32 @@ static void file_replace_cb     (GObject      *source_object,
 static void file_save_delete_cb (GObject      *source_object,
                                  GAsyncResult *result,
                                  gpointer      user_data);
+static void
+write_guint64_to_file (guint64       number,
+                       GFile        *file,
+                       GTask        *task,
+                       GCancellable *cancellable)
+{
+  union
+    {
+      guint64 u64;
+      const guint8 u8[8];
+    } number_union;
+  number_union.u64 = number;
+
+  g_autoptr(GBytes) number_bytes = g_bytes_new (number_union.u8,
```

Comment:
Strangely when I implemented this change it broke one of the unit tests so I reverted it.

Reply:
Probably because the buffer passed to `g_file_replace_contents_async` must remain valid until its callback is called, but this is a stack variable that will become invalid almost immediately.

Reply:
Do you know how it broke? Would be worth investigating a bit.

Reply:
Yeah I think Will is right about the reason it didn't work to use `g_file_replace_contents_async`. Anyway, I pushed a fixup with a simpler implementation.